icu.macro.js

```diff
@@ -137,26 +141,33 @@ function selectAsJSX(parentPath, { attributes }, babel) {
         // copy the i18nKey
         mem.attributesToCopy.push(attr.node);
       } else if (attr.node.name.name === 'switch') {
-        // take the switch for plural element
-        mem.values.push(toObjectProperty(attr.node.value.expression.name));
-        mem.defaults = `{${attr.node.value.expression.name}, select, ${mem.defaults}`;
+        // take the switch for select element
+        let exprName = attr.node.value.expression.name;
+        if (!exprName) {
+          exprName = 'selectKey';
```

Comment:
I couldn't use `switch` because it is a reserved word. I'm open to alternate names for this.

It is used when `switch` is an expression.

```
<Select
  switch={'fe'+'male'} // very contrived example
  ...
/>
// This transforms to:
<Trans values={{ selectKey: 'fe'+'male' }} defaults="{selectKey, select, ....} />
```
I don't think this will be used very often, but it was the same logic as `Plural` and `count` so I included it for completeness.

Reply:
looks ok for me